gammapy/estimators/flux_map.py

```diff
@@ -0,0 +1,480 @@
+            self.e_min, self.e_max
+        )
+        ref_dnde = self.ref_model.spectral_model(self.e_ref)
+        factor = ref_flux / ref_dnde
+        return factor[:, np.newaxis, np.newaxis]
+
+    @property
+    def flux(self):
+        return self.dnde * self._to_flux()
+
+    @property
+    def flux_err(self):
+        return self.dnde_err * self._to_flux()
+
+    @property
+    def flux_errn(self):
+        return self.dnde_errn * self._to_flux()
+
+    @property
+    def flux_errp(self):
+        return self.dnde_errp * self._to_flux()
+
+    @property
+    def flux_ul(self):
+        return self.dnde_ul * self._to_flux()
+
+    def _to_eflux(self):
+        """Conversion factor to apply to dnde-like quantities to obtain fluxes."""
+        ref_eflux = self.ref_model.spectral_model.energy_flux(
+            self.e_min, self.e_max
+        )
+        ref_dnde = self.ref_model.spectral_model(self.e_ref)
+        factor = ref_eflux / ref_dnde
+        return factor[:, np.newaxis, np.newaxis]
+
+    @property
+    def eflux(self):
+        return self.dnde * self._to_eflux()
+
+    @property
+    def eflux_err(self):
+        return self.dnde_err * self._to_eflux()
+
+    @property
+    def eflux_errn(self):
+        return self.dnde_errn * self._to_eflux()
+
+    @property
+    def eflux_errp(self):
+        return self.dnde_errp * self._to_eflux()
+
+    @property
+    def eflux_ul(self):
+        return self.dnde_ul * self._to_eflux()
+
+    def _to_e2dnde(self):
+        """Conversion factor to apply to dnde-like quantities to obtain e2dnde."""
+        factor = self.e_ref ** 2
+        return factor[:, np.newaxis, np.newaxis]
+
+    @property
+    def e2dnde(self):
+        return self.dnde * self._to_e2dnde()
+
+    @property
+    def e2dnde_err(self):
+        return self.dnde_err * self._to_e2dnde()
+
+    @property
+    def e2dnde_errn(self):
+        return self.dnde_errn * self._to_e2dnde()
+
+    @property
+    def e2dnde_errp(self):
+        return self.dnde_errp * self._to_e2dnde()
+
+    @property
+    def e2dnde_ul(self):
+        return self.dnde_ul * self._to_e2dnde()
+
+    def get_flux_points(self, coord=None):
+        """Extract flux point at a given position.
+
+        The flux points are returned in the the form of a `~gammapy.estimators.FluxPoints` object
+        (which stores the flux points in an `~astropy.table.Table`)
+
+        Parameters
+        ---------
+        coord : `~astropy.coordinates.SkyCoord`
+            the coordinate where the flux points are extracted.
+
+        Returns
+        -------
+        fluxpoints : `~gammapy.estimators.FluxPoints`
+            the flux points object
+        """
+        if coord is None:
+            coord = self.geom.center_skydir
+        energies = self.e_ref
+        coords = MapCoord.create(dict(skycoord=coord, energy=energies))
+
+        ref = self.dnde_ref.get_by_coord(coords) * self.dnde_ref.unit
```

Comment:
I guess one could call `interp_by_coord()` as well here and let users pass the interpolation order...

Reply:
Well I think it is a bit dangerous to interpolate over flux values, errors and upper limits. A priori I would rather enforce nearest neighbor .